src/csexp_rpc/csexp_rpc.mli

```diff
@@ -0,0 +1,58 @@
+(** Threaded RPC Client & Server for Csexp payloads. This RPC implementation is
+    entirely structure free and it is up to users of this library to impose some
+    sort of conventions on the protocol.
+
+    On Unix/Mac, RPC uses Unix domain sockets. On Windows, named pipes are used.
+
+    All functions reurning a fiber are non-blocking and run in their own thread. *)
+open Stdune
+
+module Scheduler : sig
+  (** Hook into the fiber scheduler. *)
+  type t =
+    { on_event : Fiber.fill -> unit  (** Called when any rpc event completes *)
+    ; register_pending_ivar : unit -> unit
+          (** Called before any ivar is created *)
+    ; thread : 'a 'b. ('a -> 'b) -> 'a -> Thread.t
+          (** We spawn threads through this function in case the scheduler wants
+              to block signals *)
+    }
+end
+
+module Session : sig
+  (** Rpc session backed by two threads. One thread for reading, and another for
+      writing *)
+  type t
+
+  val create : in_channel -> out_channel -> Scheduler.t -> t
+
+  (* [write t x] writes the s-expression when [x] is [Some sexp], and closes the
+     session if [x = None ] *)
+  val write : t -> Sexp.t option -> unit Fiber.t
```

Comment:
IMO, an explicit variant would be clearer:

```
Message of Sexp.t | End_of_session
```

or two separate functions.

Reply:
It was two separate functions before, but it made using it in conjunction with `Fiber.Stream` more annoying.

In any case, I might have to change this because of my comment [here](https://github.com/ocaml/dune/pull/3920#issuecomment-770564252)

In which case we'd need a close function like:

```
val close : [`Read | `Write] -> bool
```

Reply:
Me and Jeremie concluded that the current interface is fine, but `write None` should really mean close the write end only.